packages/react-components/babel-preset-storybook-full-source/src/fullsource.ts

```diff
@@ -30,49 +31,95 @@ export function fullSourcePlugin(babel: typeof Babel, options: BabelPluginOption
     name: PLUGIN_NAME,
     visitor: {
       // eslint-disable-next-line @typescript-eslint/naming-convention
-      VariableDeclarator(path, state) {
-        if (
-          t.isIdentifier(path.node.id) &&
-          t.isStringLiteral(path.node.init) &&
-          path.node.id.name === '__STORY__' &&
-          path.parentPath.isVariableDeclaration()
-        ) {
-          const transformedCode = babel.transformSync(path.node.init.value, {
-            ...state.file.opts,
-            compact: false,
-            retainLines: true,
-            comments: false,
-            plugins: [[modifyImportsPlugin, options], removeStorybookParameters],
-          })?.code;
-          const code = prettier.format(transformedCode ?? '', { parser: 'babel-ts' });
+      Program(programPath, state) {
```

Comment:
is this the only possible pattern, couldn't we stick with more focused node selectors ? traversing whole source file with doing recursive calles on program AST tends to be not the most performant solution.

Reply:
Updated